### PR TITLE
Fix cfn-lint breaking changes

### PR DESCRIPTION
**Pull request by viren-nadkarni**
[cfn-lint 1.3.0](https://pypi.org/project/cfn-lint/1.3.0) was released on 2024-06-19.

Currently Moto pins it as `cfn-lint>=0.40.0` which causes the latest version to be picked up and integration to break.

This PR fixes the breaking changes and updates its usage.

### CODE REVIEW

File: moto/cloudformation/utils.py [hunk 1]
```diff
@@ -59,24 +58,10 @@ def _f(loader: Any, tag: Any, node: Any) -> Any:
 
 def validate_template_cfn_lint(template: str) -> List[Any]:
     # Importing cfnlint adds a significant overhead, so we keep it local
-    from cfnlint import core, decode
-
-    # Save the template to a temporary file -- cfn-lint requires a file
-    filename = "file.tmp"
-    with open(filename, "w") as file:
-        file.write(template)
-    abs_filename = os.path.abspath(filename)
```
Comment (viren-nadkarni) [inline moto/cloudformation/utils.py:68]: Cfnlint no longer requires a file input

File: moto/cloudformation/utils.py [hunk 2]
```diff
@@ -59,24 +58,10 @@ def _f(loader: Any, tag: Any, node: Any) -> Any:
...
-
-    # decode handles both yaml and json
-    try:
-        template, matches = decode.decode(abs_filename, False)
-    except TypeError:
-        # As of cfn-lint 0.39.0, the second argument (ignore_bad_template) was dropped
-        # https://github.com/aws-cloudformation/cfn-python-lint/pull/1580
-        template, matches = decode.decode(abs_filename)
```
Comment (viren-nadkarni) [inline moto/cloudformation/utils.py:76]: The new `cfnlint.api.lint(...)` handles this itself

File: moto/cloudformation/utils.py [hunk 3]
```diff
@@ -59,24 +58,10 @@ def _f(loader: Any, tag: Any, node: Any) -> Any:
...
+    from cfnlint import api, config, core
 
     # Set cfn-lint to info
-    core.configure_logging(None)
```
Comment (viren-nadkarni) [inline moto/cloudformation/utils.py:79]: `cfnlint.core.configure_logging(...)` was removed

File: moto/cloudformation/utils.py [hunk 4]
```diff
@@ -85,9 +70,7 @@ def validate_template_cfn_lint(template: str) -> List[Any]:
     regions = ["us-east-1"]
 
     # Process all the rules and gather the errors
-    matches = core.run_checks(abs_filename, template, rules, regions)
```
Comment (viren-nadkarni) [inline moto/cloudformation/utils.py:88]: `cfnlint.core.run_checks(...)` was removed

**Comment by bblommers** [comment]
Hi @viren-nadkarni, thanks for the PR!

V1 only came out recently, so I think we want to support both v0 and v1 for now.
I'm sure there are plenty of people who want to continue using v0 for now, and I don't want to force anyone to upgrade.

I'd be happy to make the necessary changes to support both versions, we already do that for a few other dependencies (see moto/core/versions py) - I just don't know when I'll have time for that.

------

Related announcement from AWS, for future reference:
https://aws.amazon.com/blogs/devops/aws-cloudformation-linter-v1/

**Review by bblommers — APPROVED** [review]
Awesome, thank you for fixing this @viren-nadkarni!

**Pull request merged**